src/core/exec-invoke.c

```diff
@@ -3033,13 +3033,35 @@ static int pick_versions(
         return 0;
 }
 
+static bool can_apply_cgroup_namespace(const ExecContext *context, const ExecParameters *params) {
+        int r;
+
+        r = cg_all_unified();
+        if (r < 0) {
+                log_exec_warning_errno(context, params, r, "Failed to determine cgroup hierarchy version: %m");
+                return false;
+        } else if (r == 0) {
+                log_exec_warning(context, params, "ProtectControlGroups=%s is configured, but the unified cgroups hierarchy is not set up, ignoring.",
+                                 protect_control_groups_to_string(context->protect_control_groups));
+                return false;
+        }
+
+        if (!ns_type_supported(NAMESPACE_CGROUP)) {
```

Comment:
Maybe, it is better to downgrade the feature if the namespace is not supported?
private -> no
strict -> yes (read-only).

Reply:
Yup sounds good. The private downgrade kinda de facto works with the current PR but the strict will need some changes (especially as we need to bind mount and not just do a private mount since we could be on cgroupv1)